plugin/core/collections.py

```diff
@@ -0,0 +1,115 @@
+"""
+Module with additional collections.
+"""
+from .typing import Optional, Dict, Any
+
+
+class DottedDict:
+
+    __slots__ = ('_d')
```

Comment:
From https://stackoverflow.com/questions/472000/usage-of-slots:
> Requirements:
>  - To have attributes named in `__slots__` to actually be stored in slots instead of a `__dict__`, a class must inherit from object.
`

Reply:
I think that's old info. Every class implicitly inherits from `object` in python3. But there is a bug on that line... It should be `('_d',)`.

Without `__slots__`:
```
>>> from LSP.plugin.core.collections import DottedDict
>>> from sys import getsizeof
>>> getsizeof(DottedDict())
48
```

With `__slots__`:
```
>>> from LSP.plugin.core.collections import DottedDict
>>> from sys import getsizeof
>>> getsizeof(DottedDict())
40
```
What matters to me is the type-safety though.